frontend/src/components/CalcSpectrum.tsx

```diff
@@ -123,204 +138,102 @@ export const CalcSpectrum: React.FC = () => {
-      }
-    }
-
-    updatedValidationErrors.molecule = new Array(params.species.length).fill(
-      undefined
-    );
-    updatedValidationErrors.mole_fraction = new Array(
-      params.species.length
-    ).fill(undefined);
-
-    params.species.forEach((species, index) => {
-      if (!species.molecule) {
-        updatedValidationErrors.molecule[index] = "Molecule must be defined";
-      }
-      if (
-        species.mole_fraction === undefined ||
-        Number.isNaN(species.mole_fraction)
-      ) {
-        updatedValidationErrors.mole_fraction[index] =
-          "Mole fraction must be defined";
-      } else if (species.mole_fraction && species.mole_fraction < 0) {
-        updatedValidationErrors.mole_fraction[index] =
-          "Mole fraction cannot be negative";
-      }
-    });
-
-    if (Number.isNaN(params.tgas)) {
-      updatedValidationErrors.tgas = "Tgas must be defined";
-    } else if (params.tgas < 1 || params.tgas > 9000) {
-      updatedValidationErrors.tgas = "Tgas must be between 1K and 9000K";
-    } else {
-      updatedValidationErrors.tgas = undefined;
-    }
-
-    if (Number.isNaN(params.pressure)) {
-      updatedValidationErrors.pressure = "Pressure must be defined";
-    } else if (params.pressure < 0) {
-      updatedValidationErrors.pressure = "Pressure cannot be negative";
-    } else {
-      updatedValidationErrors.pressure = undefined;
-    }
-    if (params.database == "geisa") {
+  React.useEffect(() => {
+    if (databaseWatch === "geisa") {
       setUseGesia(true);
     }
-    if (params.database == "hitran") {
-      setUseGesia(false);
-    }
-    if (Number.isNaN(params.path_length)) {
-      updatedValidationErrors.path_length = "Path length must be defined";
-    } else if (params.path_length < 0) {
-      updatedValidationErrors.path_length = "Path length cannot be negative";
-    } else {
-      updatedValidationErrors.path_length = undefined;
-    }
-
-    setValidationErrors({ ...validationErrors, ...updatedValidationErrors });
-  };
-
-  const hasValidationErrors = (validationErrors: ValidationErrors): boolean =>
-    Object.values(validationErrors).some(
-      (error: string | string[] | undefined) =>
-        Array.isArray(error)
-          ? error.some((error: string | undefined) => error)
-          : error
-    );
+  }, [databaseWatch]);
 
-  //switch
   const UseNonEquilibriumCalculations = () => (
-    <FormControlLabel
-      label="Use non-equilibrium calculations"
-      control={
-        <Switch
-          //non-equ is true here
-          checked={isNonEquilibrium}
-          onChange={(e) => {
-            setIsNonEquilibrium(e.target.checked);
-            if (e.target.checked) {
-              setParams({ ...params, tvib: params.tgas, trot: params.tgas });
-            } else {
-              setParams({ ...params, tvib: null, trot: null });
-            }
-          }}
+    <Controller
+      // @ts-ignore
```

Comment:
AFAIK this ts-ignore can be fixed by adding `useNonEqi` as a field under the `FormValues` type.